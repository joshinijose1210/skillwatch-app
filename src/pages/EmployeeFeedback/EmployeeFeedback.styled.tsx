import { DateRangePicker } from '@medly-components/core';
import styled from 'styled-components';

export const StyledWrapper = styled.div`
    display: flex;
    justify-content: space-between;
`;

export const StyledDateRangePicker = styled(DateRangePicker)`
    max-width: 25rem;
    max-height: 45px;
    margin-left: 1rem;
    div {
        padding: 0.25rem;
    }
    label {
        line-height: 1.9rem;
        color: #666 !important;
    }
    input {
        font-size: 1.3rem;
        &::placeholder {
            color: transparent;
        }
    }

    svg path {
        fill: #666;
    }
`;

export const DatePickerDiv = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
    position: relative;

    svg:nth-child(2) {
        position: absolute;
        right: 1.2rem;
        z-index: 1;

        :hover {
            cursor: pointer;
        }
    }

    svg:nth-child(2) > path {
        :hover {
            fill: red;
        }
    }
`;

export const FilterContainer = styled.div`
    display: flex;
    align-items: center;
`;
