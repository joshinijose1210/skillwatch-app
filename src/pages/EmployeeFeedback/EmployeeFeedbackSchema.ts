import {
    DateFormatter,
    EmpFeedbackActionFormatter,
    FeedbackTagFormatter,
    NameWithEmpIdFormatter,
    ShowFeedbackDataFormatter
} from '@components';
import { TableColumnConfig } from '@medly-components/core';

export const EmployeeFeedbackSchema: TableColumnConfig[] = [
    {
        title: 'Date',
        field: 'date',
        component: DateFormatter,
        sortable: true,
        fraction: 0.2
    },
    {
        title: 'Feedback From',
        field: 'feedbackFromId',
        component: NameWithEmpIdFormatter,
        fraction: 0.4
    },
    {
        title: 'Feedback',
        field: 'feedbackData',
        component: ShowFeedbackDataFormatter,
        fraction: 1
    },
    {
        title: 'Feedback About',
        field: 'feedbackToId',
        component: NameWithEmpIdFormatter,
        fraction: 0.4
    },
    {
        title: 'Feedback Type',
        field: 'feedbackType',
        fraction: 0.3,
        component: FeedbackTagFormatter
    },
    { title: 'Action', field: '', component: EmpFeedbackActionFormatter, fraction: 0.2 }
];
