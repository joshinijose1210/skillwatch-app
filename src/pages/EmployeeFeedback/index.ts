import EmployeeFeedback from './EmployeeFeedback';
import ViewFeedback from './ViewFeedback';

export { EmployeeFeedback, ViewFeedback };
