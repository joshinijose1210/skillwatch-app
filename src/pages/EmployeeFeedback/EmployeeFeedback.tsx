import { StyledSearchBox, StyledSingleSelect } from '@common';
import { PageContent } from '@components';
import { CustomTable } from '@components/reusableComponents/CustomTable/CustomTable';
import ListHeader from '@components/reusableComponents/ListHeader';
import { CancelIcon } from '@medly-components/icons';
import { FilterDiv } from '@pages/CheckInWithManager/CheckInWithManager.styled';
import { DatePickerDiv, FilterContainer, StyledDateRangePicker } from './EmployeeFeedback.styled';
import { EmployeeFeedbackSchema } from './EmployeeFeedbackSchema';
import { useEmployeeFeedback } from './useEmployeeFeedback';

const EmployeeFeedback = () => {
    const {
        onSearchClear,
        handleDateChange,
        handlePageChange,
        data,
        isError,
        isLoading,
        ActivePage,
        searchText,
        handleDatePickerClear,
        tagsList,
        handleDropdownChange,
        handleOrderChange,
        search,
        onSearchChange,
        feedbackFilterData
    } = useEmployeeFeedback();

    return (
        <PageContent>
            <ListHeader title="Feedback" />

            <FilterDiv>
                <StyledSearchBox
                    onInputChange={onSearchChange}
                    isLoading={search.searchText !== '' && isLoading}
                    onClear={onSearchClear}
                    value={searchText}
                    size="M"
                    minWidth="25rem"
                    placeholder="Search Employee"
                    data-testid="search-employee"
                />
                <FilterContainer>
                    <StyledSingleSelect
                        options={tagsList}
                        label="Feedback Type"
                        placeholder="Select Feedback Type"
                        variant="outlined"
                        size="M"
                        onChange={value => handleDropdownChange(value)}
                        value={feedbackFilterData.feedbackType}
                    />

                    <DatePickerDiv>
                        <StyledDateRangePicker
                            displayFormat={'dd-MM-yyyy'}
                            variant="outlined"
                            size="M"
                            value={feedbackFilterData.dateRange}
                            onChange={value => handleDateChange(value)}
                            maxSelectableDate={new Date()}
                            popoverPlacement={'bottom-end'}
                            autoComplete="off"
                            data-testid="date-picker"
                        />
                        {feedbackFilterData.dateRange &&
                            (feedbackFilterData.dateRange.startDate || feedbackFilterData.dateRange.endDate) && (
                                <CancelIcon onClick={handleDatePickerClear} />
                            )}
                    </DatePickerDiv>
                </FilterContainer>
            </FilterDiv>

            <CustomTable
                data={isError ? [] : data?.feedbacks || []}
                columns={EmployeeFeedbackSchema}
                isLoading={isLoading}
                activePage={parseInt(ActivePage || '1')}
                handlePageChange={handlePageChange}
                count={data?.totalFeedbacks || 0}
                setSortOrder={handleOrderChange}
            />
        </PageContent>
    );
};

export default EmployeeFeedback;
