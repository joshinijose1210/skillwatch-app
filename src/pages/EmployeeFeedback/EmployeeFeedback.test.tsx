import { fireEvent, RTKRender, screen, waitFor } from '@utils/test-utils';
import { EmployeeFeedbackMockData } from '@mocks/employeefeedback';
import { apiInstance } from '@utils';
import MockAdapter from 'axios-mock-adapter';
import EmployeeFeedback from './EmployeeFeedback';
import { routeConstants } from '@constants';

const mock = new MockAdapter(apiInstance, { onNoMatch: 'throwException' });
const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...(jest.requireActual('react-router-dom') as any),
    useNavigate: () => mockNavigate
}));

beforeAll(() => {
    const { feedbackTypeData } = EmployeeFeedbackMockData.get;
    mock.onGet(feedbackTypeData.url).replyOnce(feedbackTypeData.responseCode, feedbackTypeData.data);
});

describe('Employee Feedback', () => {
    it('should render properly', () => {
        const { initialData, feedbackTypeData } = EmployeeFeedbackMockData.get;
        mock.onGet(initialData.url)
            .replyOnce(initialData.responseCode, initialData.data)
            .onGet(feedbackTypeData.url)
            .replyOnce(feedbackTypeData.responseCode, feedbackTypeData.data);
        const { container } = RTKRender(<EmployeeFeedback />);
        expect(container).toMatchSnapshot();
    });

    it('should test Search Employee', async () => {
        const { initialData, searchEmployeeFromData } = EmployeeFeedbackMockData.get;
        mock.onGet(initialData.url)
            .replyOnce(initialData.responseCode, initialData.data)
            .onGet(searchEmployeeFromData.url)
            .replyOnce(searchEmployeeFromData.responseCoe, searchEmployeeFromData.data);
        const { container } = RTKRender(<EmployeeFeedback />);
        const initialResult = await screen.findAllByText('Rushad Eliyas Shaikh (SR0051)');
        expect(initialResult.length).toEqual(2);
        const searchEmployee = screen.getByTestId('search-employee');
        const searchBtn = screen.getByTitle('search icon');
        fireEvent.change(searchEmployee, { target: { value: 'Moly Agarwal' } });
        fireEvent.click(searchBtn);
        const searchResult = await screen.findAllByText('Moly Agarwal (SR0050)');
        expect(searchResult.length).toEqual(1);
        await waitFor(() => expect(initialResult.length).toEqual(2), { timeout: 3000 });
        const searchClearIcon = container.querySelector('[title="close icon"]');
        searchClearIcon && fireEvent.click(searchClearIcon);
        await waitFor(() => expect(mockNavigate).toHaveBeenCalledWith(`${routeConstants.employeeFeedback}/1`), { timeout: 3000 });
    });

    it('should test sorting functionality', async () => {
        const { initialData } = EmployeeFeedbackMockData.get;
        mock.onGet(initialData.url)
            .replyOnce(initialData.responseCode, initialData.data)
            .onGet(initialData.sortUrl)
            .replyOnce(initialData.responseCode, initialData.data);

        RTKRender(<EmployeeFeedback />);

        await waitFor(() => screen.findAllByText('View'), { timeout: 3000 });
        const spanElement = screen.getByText('Date');
        fireEvent.click(spanElement);
        await waitFor(() => expect(screen.getAllByText('View')[0]).toBeInTheDocument(), { timeout: 3000 });
    });

    it('should test datepicker', async () => {
        const { dateBasedTestData } = EmployeeFeedbackMockData.get;
        mock.onGet(dateBasedTestData.url).replyOnce(dateBasedTestData.responseCode, dateBasedTestData.data);
        const { container } = RTKRender(<EmployeeFeedback />);
        const datepicker = screen.getByTitle('medly-date-range-picker-calendar-icon');
        fireEvent.click(datepicker);

        await waitFor(() => container.querySelector('#review-cycle-calendar-months-wrapper'));
        const searchEmployee = screen.getByTestId('search-employee');
        fireEvent.change(searchEmployee, { target: { value: 'Rushad' } });
        fireEvent.change(screen.getAllByPlaceholderText('DD - MM - YYYY')[0], { target: { value: '11 - 10 - 2023' } }); // From
        fireEvent.change(screen.getAllByPlaceholderText('DD - MM - YYYY')[1], { target: { value: '15 - 10 - 2023' } }); // To

        expect(await screen.findByText('11/10/2023')).toBeInTheDocument();
        expect(await screen.findByText('14/10/2023')).toBeInTheDocument();
        fireEvent.change(searchEmployee, { target: { value: '' } });
        fireEvent.change(screen.getAllByPlaceholderText('DD - MM - YYYY')[0], { target: { value: '15 - 10 - 2023' } }); // From
        fireEvent.change(screen.getAllByPlaceholderText('DD - MM - YYYY')[1], { target: { value: '11 - 10 - 2023' } }); // To

        const dateClearIcon = container.querySelector('[data-testid="date-picker"] + svg');
        dateClearIcon && fireEvent.click(dateClearIcon);
        await waitFor(() => expect(screen.queryByText('11/10/2023')).not.toBeInTheDocument(), { timeout: 3000 });
    });

    it('should test pagination', async () => {
        const { repeatedData, paginatedData } = EmployeeFeedbackMockData.get;
        mock.onGet(repeatedData.url)
            .replyOnce(repeatedData.responseCode, repeatedData.data)
            .onGet(paginatedData.url)
            .replyOnce(paginatedData.responseCode, paginatedData.data);
        RTKRender(<EmployeeFeedback />);

        const prevResult = await screen.findAllByText('Random Name (SR0051)');
        expect(prevResult.length).toEqual(11);
        const pagnBtn = await screen.findAllByText('2');
        fireEvent.click(pagnBtn[pagnBtn.length - 1]);
        await waitFor(() => screen.findByText('Random Name 2 (SR0052)'), { timeout: 3000 });
        const currentResult = screen.getAllByText('Random Name 2 (SR0052)');
        expect(currentResult.length).toEqual(1);
    });

    it('should test redirection when clicked on View', async () => {
        const { initialData, viewFeedbackState } = EmployeeFeedbackMockData.get;
        mock.onGet(initialData.url).replyOnce(initialData.responseCode, initialData.data);

        RTKRender(<EmployeeFeedback />);

        await waitFor(() => screen.findAllByText('View'), { timeout: 3000 });
        fireEvent.click(screen.getAllByText('View')[0]);

        expect(mockNavigate).toHaveBeenCalledWith('/reports/feedback/undefined/view-feedback', {
            state: viewFeedbackState
        });
    });
});
