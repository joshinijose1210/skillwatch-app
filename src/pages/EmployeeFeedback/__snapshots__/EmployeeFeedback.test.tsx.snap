// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employee Feedback should render properly 1`] = `
<div>
  <main
    class="sc-1ehavxz-0 dDyhBJ"
  >
    <div
      class="sc-h70dbj-0 fQmYvl row expanded space-between align-center marginBottom-2"
    >
      <div
        class="sc-w8e7z6-2"
      >
        <h3
          class="sc-1ntgl3l-0 kcLBOi sc-w8e7z6-0 ibnmt"
          textalign="initial"
          textcolor="inherit"
          textvariant="h3"
          textweight="Medium"
        >
          Feedback
        </h3>
      </div>
    </div>
    <div
      class="sc-r36uql-0 kCYWST"
    >
      <div
        class="sc-1cnun24-0 irVHrL sc-tkiih6-15 cvCOyL"
        maxwidth="100%"
        minwidth="25rem"
      >
        <input
          class="sc-9l9lu7-0 lfgaXg"
          data-testid="search-employee"
          placeholder="Search Employee"
          value=""
        />
        <span
          class="sc-1lts63a-1 lckNMu"
        >
          <svg
            class="sc-1pf3x8a-0 ivKmVS"
            fill="none"
            height="1em"
            title="search icon"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M9.063 3.125a5.938 5.938 0 1 0 0 11.875 5.938 5.938 0 0 0 0-11.875zM1.875 9.063a7.187 7.187 0 1 1 12.692 4.621l3.375 3.374a.625.625 0 1 1-.884.884l-3.374-3.374A7.188 7.188 0 0 1 1.875 9.063z"
              fill="#606878"
              fill-rule="evenodd"
              stroke="#606878"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width=".3"
            />
          </svg>
        </span>
      </div>
      <div
        class="sc-18u9mah-3 kyunCi"
      >
        <div
          class="sc-1hcsztp-0 WJBGn sc-tkiih6-13 cecHDR"
          id="feedback type-wrapper"
          minwidth="20rem"
          variant="outlined"
        >
          <div
            class="sc-17d4s39-0 fGqzZg"
            id="feedback type-input-wrapper"
            minwidth="20rem"
          >
            <div
              class="sc-19px6zp-0 BrSGZ"
              filled="[object Object]"
              fusion="[object Object]"
              minrows="1"
              outlined="[object Object]"
              textvariant="[object Object]"
              variant="outlined"
            >
              <div
                class="sc-lhr232-0 btNOKw"
                variant="outlined"
              >
                <input
                  aria-describedby="feedback type-helper-text"
                  autocomplete="off"
                  class="sc-hb99wz-0 jaEbhp"
                  errortext=""
                  id="feedback type-input"
                  inputmode="none"
                  inputsize="M"
                  placeholder="Select Feedback Type"
                  type="text"
                  value=""
                  variant="outlined"
                />
                <label
                  class="sc-1du51u0-0 eMZrST"
                  for="feedback type-input"
                  inputwidth="0"
                  variant="outlined"
                >
                  Feedback Type
                </label>
              </div>
              <svg
                class="sc-1pf3x8a-0 klDMsE sc-dynlvq-0 bdmviR"
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12.792 14.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
                  fill="#000"
                />
              </svg>
            </div>
            
          </div>
        </div>
        <div
          class="sc-18u9mah-2 kXETkJ"
        >
          <div
            class="sc-17d4s39-0 bVMFZm sc-18u9mah-1 fYEEB"
            data-testid="date-picker"
            id="medly-date-range-picker-wrapper"
            minwidth="33.8rem"
          >
            <div
              class="sc-19px6zp-0 sc-13bf4az-0 BrSGZ ensfuF"
              filled="[object Object]"
              fusion="[object Object]"
              id="medly-date-range-picker-textFields"
              outlined="[object Object]"
              textvariant="[object Object]"
              variant="outlined"
            >
              <svg
                class="sc-1pf3x8a-0 ivKmVS sc-6rsprq-0 lcrXdL"
                fill="none"
                height="1em"
                id="medly-date-range-picker-calendar-icon"
                title="medly-date-range-picker-calendar-icon"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M19 4h-1V3c0-.55-.45-1-1-1s-1 .45-1 1v1H8V3c0-.55-.45-1-1-1s-1 .45-1 1v1H5c-1.11 0-1.99.9-1.99 2L3 20a2 2 0 0 0 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 15c0 .55-.45 1-1 1H6c-.55 0-1-.45-1-1V9h14v10zM9 13v-2H7v2h2zm2-2h2v2h-2v-2zm6 2v-2h-2v2h2z"
                  fill="#000"
                  fill-opacity=".54"
                  fill-rule="evenodd"
                />
              </svg>
              <div
                class="sc-lhr232-0 btNOKw"
                id="medly-date-range-picker-startDate-input-wrapper"
                variant="outlined"
              >
                <input
                  autocomplete="off"
                  class="sc-hb99wz-0 jaEbhp"
                  errortext=""
                  id="medly-date-range-picker-startDate-input"
                  inputsize="M"
                  name="START_DATE"
                  pattern="\\d{2} - \\d{2} - \\d{4}"
                  placeholder="DD - MM - YYYY"
                  value=""
                  variant="outlined"
                />
                <span
                  class="sc-1r5xr2a-0 cOYtDY"
                  variant="outlined"
                >
                  DD - MM - YYYY
                </span>
                <label
                  class="sc-1du51u0-0 eMZrST"
                  for="medly-date-range-picker-startDate-input"
                  inputwidth="0"
                  variant="outlined"
                >
                  From
                </label>
              </div>
              <i
                class="sc-yb1wrg-0 fHdWbA"
                variant="outlined"
              />
              <div
                class="sc-lhr232-0 btNOKw"
                id="medly-date-range-picker-endDate-input-wrapper"
                variant="outlined"
              >
                <input
                  autocomplete="off"
                  class="sc-hb99wz-0 jaEbhp"
                  errortext=""
                  id="medly-date-range-picker-endDate-input"
                  inputsize="M"
                  name="END_DATE"
                  pattern="\\d{2} - \\d{2} - \\d{4}"
                  placeholder="DD - MM - YYYY"
                  value=""
                  variant="outlined"
                />
                <span
                  class="sc-1r5xr2a-0 cOYtDY"
                  variant="outlined"
                >
                  DD - MM - YYYY
                </span>
                <label
                  class="sc-1du51u0-0 eMZrST"
                  for="medly-date-range-picker-endDate-input"
                  inputwidth="0"
                  variant="outlined"
                >
                  To
                </label>
              </div>
            </div>
            <span
              class="sc-avlt8d-0 llxXA-d"
              id="medly-date-range-picker-helper-text"
              variant="outlined"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-8tbhpn-3"
    >
      <div
        class="sc-1msmxxr-0 iWEzjW"
      />
      <table
        actions=""
        class="sc-1msmxxr-1 jwFsPF sc-8tbhpn-2 kCDmJI"
        columns="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
        defaultactivepage="1"
        defaultsortorder="desc"
        itemsperpage="10"
        rowclickdisablekey=""
        tabindex="-1"
        totalitems="0"
      >
        <thead
          class="sc-uxzsxp-0 haNltq"
        >
          <tr
            class="sc-1jr91bg-0 hJzMcO"
            gridtemplatecolumns=" minmax(100px, 0.2fr) minmax(100px, 0.4fr) minmax(100px, 1fr) minmax(100px, 0.4fr) minmax(100px, 0.3fr) minmax(100px, 0.2fr)"
            style="grid-template-columns: minmax(100px, 0.2fr) minmax(100px, 0.4fr) minmax(100px, 1fr) minmax(100px, 0.4fr) minmax(100px, 0.3fr) minmax(100px, 0.2fr);"
          >
            <th
              class="sc-hjmm4m-0 iOyNiA"
              tablesize="M"
            >
              <button
                class="sc-hjmm4m-2 kpjakb"
                tablesize="M"
              >
                <span
                  class="sc-1ntgl3l-0 ghpJwn"
                  textalign="initial"
                  textcolor="inherit"
                >
                  Date
                </span>
                <svg
                  class="sc-1pf3x8a-0 bMQNVa"
                  fill="none"
                  height="1em"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.229 6.935a1 1 0 0 1 1.543 0l2.208 2.679a1 1 0 0 1-.77 1.636H9.79a1 1 0 0 1-.77-1.636l2.209-2.679zM12.771 17.065a1 1 0 0 1-1.543 0L9.02 14.386a1 1 0 0 1 .772-1.636h4.418a1 1 0 0 1 .771 1.636l-2.209 2.679z"
                    fill="#435465"
                  />
                </svg>
              </button>
              <span
                class="sc-hjmm4m-1 cuEDpa"
              />
            </th>
            <th
              class="sc-hjmm4m-0 iOyNiA"
              tablesize="M"
            >
              <button
                class="sc-hjmm4m-2 kpjakb"
                tablesize="M"
              >
                <span
                  class="sc-1ntgl3l-0 ghpJwn"
                  textalign="initial"
                  textcolor="inherit"
                >
                  Feedback From
                </span>
              </button>
              <span
                class="sc-hjmm4m-1 cuEDpa"
              />
            </th>
            <th
              class="sc-hjmm4m-0 iOyNiA"
              tablesize="M"
            >
              <button
                class="sc-hjmm4m-2 kpjakb"
                tablesize="M"
              >
                <span
                  class="sc-1ntgl3l-0 ghpJwn"
                  textalign="initial"
                  textcolor="inherit"
                >
                  Feedback
                </span>
              </button>
              <span
                class="sc-hjmm4m-1 cuEDpa"
              />
            </th>
            <th
              class="sc-hjmm4m-0 iOyNiA"
              tablesize="M"
            >
              <button
                class="sc-hjmm4m-2 kpjakb"
                tablesize="M"
              >
                <span
                  class="sc-1ntgl3l-0 ghpJwn"
                  textalign="initial"
                  textcolor="inherit"
                >
                  Feedback About
                </span>
              </button>
              <span
                class="sc-hjmm4m-1 cuEDpa"
              />
            </th>
            <th
              class="sc-hjmm4m-0 iOyNiA"
              tablesize="M"
            >
              <button
                class="sc-hjmm4m-2 kpjakb"
                tablesize="M"
              >
                <span
                  class="sc-1ntgl3l-0 ghpJwn"
                  textalign="initial"
                  textcolor="inherit"
                >
                  Feedback Type
                </span>
              </button>
              <span
                class="sc-hjmm4m-1 cuEDpa"
              />
            </th>
            <th
              class="sc-hjmm4m-0 iOyNiA"
              tablesize="M"
            >
              <button
                class="sc-hjmm4m-2 kpjakb"
                tablesize="M"
              >
                <span
                  class="sc-1ntgl3l-0 ghpJwn"
                  textalign="initial"
                  textcolor="inherit"
                >
                  Action
                </span>
              </button>
              <span
                class="sc-hjmm4m-1 cuEDpa"
              />
            </th>
          </tr>
        </thead>
        <tbody
          class="sc-c7cniy-0 kShdfA"
        >
          <tr
            class="sc-1yxujaa-0 bgyIwA"
            gridtemplatecolumns=" minmax(100px, 0.2fr) minmax(100px, 0.4fr) minmax(100px, 1fr) minmax(100px, 0.4fr) minmax(100px, 0.3fr) minmax(100px, 0.2fr)"
            style="grid-template-columns: minmax(100px, 0.2fr) minmax(100px, 0.4fr) minmax(100px, 1fr) minmax(100px, 0.4fr) minmax(100px, 0.3fr) minmax(100px, 0.2fr);"
            tabindex="0"
          >
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
          </tr>
          <tr
            class="sc-1yxujaa-0 bgyIwA"
            gridtemplatecolumns=" minmax(100px, 0.2fr) minmax(100px, 0.4fr) minmax(100px, 1fr) minmax(100px, 0.4fr) minmax(100px, 0.3fr) minmax(100px, 0.2fr)"
            style="grid-template-columns: minmax(100px, 0.2fr) minmax(100px, 0.4fr) minmax(100px, 1fr) minmax(100px, 0.4fr) minmax(100px, 0.3fr) minmax(100px, 0.2fr);"
            tabindex="1"
          >
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
          </tr>
          <tr
            class="sc-1yxujaa-0 bgyIwA"
            gridtemplatecolumns=" minmax(100px, 0.2fr) minmax(100px, 0.4fr) minmax(100px, 1fr) minmax(100px, 0.4fr) minmax(100px, 0.3fr) minmax(100px, 0.2fr)"
            style="grid-template-columns: minmax(100px, 0.2fr) minmax(100px, 0.4fr) minmax(100px, 1fr) minmax(100px, 0.4fr) minmax(100px, 0.3fr) minmax(100px, 0.2fr);"
            tabindex="2"
          >
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
          </tr>
          <tr
            class="sc-1yxujaa-0 bgyIwA"
            gridtemplatecolumns=" minmax(100px, 0.2fr) minmax(100px, 0.4fr) minmax(100px, 1fr) minmax(100px, 0.4fr) minmax(100px, 0.3fr) minmax(100px, 0.2fr)"
            style="grid-template-columns: minmax(100px, 0.2fr) minmax(100px, 0.4fr) minmax(100px, 1fr) minmax(100px, 0.4fr) minmax(100px, 0.3fr) minmax(100px, 0.2fr);"
            tabindex="3"
          >
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
          </tr>
          <tr
            class="sc-1yxujaa-0 bgyIwA"
            gridtemplatecolumns=" minmax(100px, 0.2fr) minmax(100px, 0.4fr) minmax(100px, 1fr) minmax(100px, 0.4fr) minmax(100px, 0.3fr) minmax(100px, 0.2fr)"
            style="grid-template-columns: minmax(100px, 0.2fr) minmax(100px, 0.4fr) minmax(100px, 1fr) minmax(100px, 0.4fr) minmax(100px, 0.3fr) minmax(100px, 0.2fr);"
            tabindex="4"
          >
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </main>
</div>
`;
